third_party/java/arcs/build_defs/internal/kotlin.bzl

```diff
@@ -706,6 +729,7 @@ def arcs_kt_schema(
                 language_name = "Kotlin",
                 wasm = wasm,
                 test_harness = False,
+                type_slicing = KOTLIN_ENABLE_TYPE_SLICING,
```

Comment:
I could be mistaken, but it looks like setting the particle_type_slicing flag to READY will effectively "launch" this feature for all clients, without them actually opting in. Is that right?

Reply:
It does, because this is a compilation-time thing. I recently made a change such that read/write handles are defined as `ReadWriteCollectionHandle<MyEntity, MyEntitySlice>`, but the `MyEntitySlice` is a typealias to `MyEntity`. This CL makes it the same with the flag off, or switches to the actual slice interface with the flag on.

While we could have codegen producing the slice interfaces but not using them, you can't switch the typealias at runtime.

Reply:
Something else to consider: in the current code, this only thing this feature really changes is what types are used as type params in generics, which are erased at runtime. So even with the flag turned on, there's no logical changes in prod - if it compiles (i.e. the interfaces line up correctly), it works.

Once we do launch and clients start using it (by transparently passing sliceable entity types), then turning the flag off will cause them to fail compilation. I can't think of any way out of this other than moving all such code into their own, separate build targets guarded by the flag, and that's an incredibly intrusive requirement.